server/UserAuth.php

```diff
@@ -136,6 +136,24 @@ public function addCap(string $cap): void
         $this->user->add_cap($cap);
     }
 
+    public function removeUserCapabilities(?WP_User $user = null): void
+    {
+        $user = $user ?? $this->user;
+
+        $user->remove_cap(self::CAP_MANAGE_AGENTWP_CONNECTION);
+        $user->remove_cap(self::CAP_MANAGE_AGENTWP_USERS);
+        $user->remove_cap(self::CAP_AGENTWP_ACCESS);
+    }
+
+    public function removeCapabilitiesFromAllUsers(): void
+    {
+        $users = get_users();
```

Comment:
this should query the users with the actual capabilities that need to be removed or execute a more efficient update. For sites with 1000s of users this will make deactivating an issue

Reply:
Problem resolved